### PR TITLE
Close any previous generated keys ResultSet before assigning a new value

### PR DESCRIPTION
**Pull request by jesperpedersen**

### CODE REVIEW

File: src/main/java/com/impossibl/postgres/jdbc/PGStatement.java [hunk 1]
```diff
@@ -227,6 +228,11 @@ static void closeResultSets(List<WeakReference<PGResultSet>> resultSets) {
    */
   void closeResultSets() throws SQLException {
     closeResultSets(activeResultSets);
+
```
Comment (kdubb) [inline src/main/java/com/impossibl/postgres/jdbc/PGStatement.java:231]: This is unnecessary as generatedKeysResultSet is added to activeResultSets when created

File: src/main/java/com/impossibl/postgres/jdbc/PGPreparedStatement.java [hunk 2]
```diff
@@ -208,6 +208,10 @@ public boolean execute() throws SQLException {
     }
 
     if (wantsGeneratedKeys) {
+      if (generatedKeysResultSet != null && !generatedKeysResultSet.isClosed()) {
+        generatedKeysResultSet.internalClose();
+      }
+
```
Comment (kdubb) [inline src/main/java/com/impossibl/postgres/jdbc/PGPreparedStatement.java]: This needs to remove generatedKeysResultSet from the activeResultSets list as well.

File: src/main/java/com/impossibl/postgres/jdbc/PGPreparedStatement.java [hunk 3]
```diff
@@ -318,6 +322,10 @@ public void clearBatch() throws SQLException {
 
       }
 
+      if (generatedKeysResultSet != null && !generatedKeysResultSet.isClosed()) {
+        generatedKeysResultSet.internalClose();
+      }
+
```
Comment (kdubb) [inline src/main/java/com/impossibl/postgres/jdbc/PGPreparedStatement.java]: This code should be removed.  executeBatch is missing a call to closeResultSets, which would close all activeResultSets, at the outset of the method; which all executeXXX methods are required to do.  generatedKeysResultSet is in the activeResultSets list.

**Comment by jesperpedersen** [comment]
Fixed

**Pull request merged**